### PR TITLE
Ignore commands from being considered if they have no cluster criteria

### PR DESCRIPTION
**Pull request by tgianos**
Upgrade gradle and plugins as well

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,7 +57,7 @@ configure(javaProjects) {
     apply plugin: "java"
     apply plugin: "checkstyle"
     apply plugin: "com.github.spotbugs"
-    apply plugin: "pmd"
+//    apply plugin: "pmd"
```
Comment (mprimi) [inline build.gradle:60]: I haven't noticed this difference.

File: genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/queries/specifications/JpaCommandSpecs.java [hunk 2]
```diff
@@ -114,5 +114,8 @@ private JpaCommandSpecs() {
                 CommandEntity_.id,
                 criterion
             );
+
+            return cb.and(predicate, cb.isNotEmpty(root.get(CommandEntity_.clusterCriteria)));
```
Comment (mprimi) [inline genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/queries/specifications/JpaCommandSpecs.java:118]: Would this be more efficient if the order was swapped?

File: build.gradle (hunk 1 above)
Comment (tgianos) [reply build.gradle]: For now I pulled this out

File: genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/queries/specifications/JpaCommandSpecs.java (hunk 2 above)
Comment (tgianos) [reply genie-web/src/main/java/com/netflix/genie/web/data/services/impl/jpa/queries/specifications/JpaCommandSpecs.java:118]: As discussed offline this is unknown to me due to the layers the path would go through to generate the sql as well as if it's better to short circuit on a join clause vs an in row clause first. For now we'll try this and if we see a performance problem we can try switching it.

**Pull request merged**